automation_use_cases/scan_automation/README.md

```diff
@@ -0,0 +1,73 @@
+# Scan Automation
+
+## Getting Started
+The Scan Automation solution is designed to automate the triggering and reporting of a scan in InsightAppSec.
+
+The steps for the scan automation process are executed as follows:
+
+1. Grab application and scan config information from configuration file
+2. Lookup and store IDs for application and scan config
+3. Trigger the scan using IDs
+4. Track status of the scan
+5. Report on scan findings
+
+### Configuration
+
+To begin the process of scan automation there must be a target application and scan config defined in the
+`config/settings.yml` file. The `scan_info` section houses this information. The `app_name` field will hold
+the name of the InsightAppSec application to scan. The `config` field will hold the ID of the InsightAppSec
+scan config to use for the scan. Below is an example of a valid `scan_info` configuration. The `status_check_interval` field represents the number of seconds between each status check while tracking the scan's progress. There is a minimum value of 60 seconds required. If left unspecified, the interval defaults to 300 seconds.
+
+```
+scan_info:
+  app_name: MyApplication
+  config: Default
+  status_check_interval: 300
+```
+
+Connection settings must also be defined to connect to InsightAppSec and perform the required operations. Under 
+`connection` in `config/settings.yml`, there are fields for both `region` and `api_key` to facilitate this 
+connection.
+
+```
+connection:
+  region: us
+  api_key: 
+```
+
+Rapid7 recommends the encryption and secure storage of sensitive values such as the API key to adhere to best 
+security practices. If an API key is not entered in `settings.yml`, then the script will prompt the user to 
+enter one upon executing it.
+
+## Usage
+
+### Running the Script
+
+The main script can be executed by navigating to the project's `bin` directory and entering the CLI command 
+`python main.py`. This will launch the script and begin creating and configuring new applications based on the 
+settings defined in `settings.yml`.
+
+### Output
+
+The script is designed to include logging as its logic is executed, written to a log file located in the `log` 
+directory. These logs contain information about each operation being performed in the scan automation process, 
+such as scan creation, scan status updates, and scan results.
+
+```
+2019-11-21 12:01:41.517277 - INFO - Region: us
+2019-11-21 12:01:41.517277 - INFO - Launched new scan: 0000000-0000-0000-0000-000000000000
+2019-11-21 12:01:41.517277 - INFO - Scan status: PENDING
+2019-11-21 12:01:41.517277 - INFO - Scan status: QUEUED
+2019-11-21 12:01:41.517277 - INFO - Scan status: QUEUED
+2019-11-21 12:01:41.517277 - INFO - Scan status: QUEUED
+2019-11-21 12:01:41.517277 - INFO - Scan status: QUEUED
+2019-11-21 12:01:41.517277 - INFO - Scan status: QUEUED
+2019-11-21 12:01:41.517277 - INFO - Scan status: QUEUED
+2019-11-21 12:01:41.517277 - INFO - Scan status: QUEUED
```

Comment:
I'd remove most of the `QUEUED` lines so it doesn't look like we force the scan to wait for ages.